File: java/20. Valid Parentheses.java
import java.util.*;
public class parenthesis_checker {
    static boolean ispar(String x) {
        Stack<Character> st = new Stack();
        for (int i = 0; i < x.length(); i++) {
            char ch = x.charAt(i);
            if (ch == '(' || ch == '{' || ch == '[') {
                st.push(ch);
                continue;
            }
            if (st.isEmpty()) {
                return false;
            }
            char top = 0;
            switch (ch) {
                case ')':
                    top = st.pop();
                    if (top == '[' || top == '{')
                        return false;
                    break;
                case ']':
                    top = st.pop();
                    if (top == '{' || top == '(')
                        return false;
                    break;
                case '}':
                    top = st.pop();
                    if (top == '[' || top == '(')
                        return false;
                    break;
            }
        }
        return st.isEmpty();
    }
    public static void main(String[] args) {
  String str = "{([])}";
  boolean s = ispar(str);
        System.out.println(s);
    }
}

File: py/20. Valid Parentheses.py
class Solution(object):
    def isValid(self, s):
        l=list(str(s))
        n=len(l)
    
        d={')':'(','}':'{',']':'['}
        l1=[]
        for i in l:
            if len(l1)==0 and i in ')]}':
                return False
            if i in '({[':
                l1.append(i)
            if len(l1)!=0 and i in ')]}':
                if d[i]==l1[-1]:
                    l1.pop()
                else:
                    return False
        if len(l1)==0:
            return True
        else:
            return False
            
        
            
                
            
       
